src/compression.rs

```diff
@@ -36,33 +38,19 @@ use crate::{
     Result,
 };
 
-/// Contains a fixed list of common text-based MIME types in order to apply compression.
-pub const TEXT_MIME_TYPES: [&str; 24] = [
-    "text/html",
-    "text/css",
-    "text/javascript",
-    "text/xml",
-    "text/plain",
-    "text/csv",
-    "text/calendar",
-    "text/markdown",
-    "text/x-yaml",
-    "text/x-toml",
-    "text/x-component",
-    "application/rtf",
-    "application/xhtml+xml",
-    "application/javascript",
-    "application/x-javascript",
-    "application/json",
-    "application/xml",
-    "application/rss+xml",
-    "application/atom+xml",
-    "font/truetype",
-    "font/opentype",
-    "application/vnd.ms-fontobject",
-    "image/svg+xml",
-    "application/wasm",
-];
+lazy_static! {
+    /// Contains a fixed list of common text-based MIME types that aren't recognizable in a generic way.
+    static ref TEXT_MIME_TYPES: HashSet<&'static str> = [
+        "application/rtf",
+        "application/javascript",
+        "application/json",
+        "application/xml",
+        "font/ttf",
+        "application/font-sfnt",
+        "application/vnd.ms-fontobject",
+        "application/wasm",
```

Comment:
Comment: the list is fine. But we could extend it a bit to other text or even some binary types in another PR.
E.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
For example, `font/woff` is missing and some others.

Reply:
I thought about `font/woff` already. But this format is compressed already. Most of the others are either uncommon on the web (`image/bmp`) or also compressed.

Reply:
> I thought about `font/woff` already.

Are you sure? Because I saw that Cloudflare compresses them for example (e.g. brotli or gzip) with some kb's gains.
See https://developers.cloudflare.com/speed/optimization/content/brotli/content-compression/

Btw we could guide us from its content types list for a future improvement.

Reply:
I don’t know what Cloudflare does and why but WOFF is definitely compressed. From https://www.w3.org/TR/WOFF20ER/:

> Web Open Font Format (WOFF) 2.0 is a proposed update to the existing [WOFF 1.0](https://www.w3.org/TR/WOFF/) with improved compression.

In fact, WOFF 2.0 uses the same Brotli compression.